### ISSUE TITLE
Multiple `Option` types must be defined explicitly in abi function definition

### ISSUE THREAD

**Issue by bitzoic** [FuelLabs/fuels-rs#420]
Having multiple different `Option`s in a Sway contract where one `Option` is defined explicitly in a function and the other is not causes an error. 

In this example, we have: 
1. A `struct` that contains an `Option` of type `Identity` 
2. Another `Option` of type `u64`

Only the `Option` of type `u64` is explicitly defined in a function, the `test_function` function. When attempting to create a `TestStruct` in the harness file to pass as an input to `input_function`, the `Option` of type `Identity` is not recognized. 

While this example only shows `Option` of type `u64` returned, the same error occurs when the `Option` of type `u64` is an input variable. Thus, the `Option` must be explicitly defined in a function.

Example Contract:

```
struct TestStruct {
    option_identity: Option<Identity>
}

abi MyContract {
    fn input_function(myStruct: TestStruct);
    fn test_function() -> Option<u64>;
}

impl MyContract for Contract {

    // Here we are taking in a struct that has an Option defined within itself, 
    // but the Option is not defined in the function definition

    fn input_function(myStruct: TestStruct) {
        // Do something here
    }

    // Here we are returning an Option and the Option is defined in the 
    // function definition

    fn test_function() -> Option<u64> {
        Option::Some(1)
    }
}
```

Corresponding harness.rs file:

```
abigen!(MyContract, "out/debug/test_project-abi.json");

mod test {

    use super::*;

    // This function works as intended

    async fn test_function(contract: &MyContract) -> Option {
        contract.test_function().call().await.unwrap().value
    }

    // This function will throw us an error

    async fn input_function(contract: &MyContract, wallet: &LocalWallet) -> CallResponse<()> {
        let identity = Identity::Address(wallet.address());

        let my_struct = TestStruct {
            option_identity: Option::Some(identity) // This line is the problem
        };

        contract.input_function(my_struct).call().await.unwrap()
    }
}
```

The corresponding error:
![optionError](https://user-images.githubusercontent.com/54727135/175077915-ca49d2fe-c8e1-4e4b-8ec1-703063e47dff.png)

**Comment by digorithm** [comment]
This should be resolved by https://github.com/FuelLabs/fuels-rs/issues/415.

**Comment by bitzoic** [comment]
> This should be resolved by #415.

I believe #415 is the interaction between Rust's `Option` and Sway's `Option` while this is purely Sway's `Option`. The same applies to #421

**Comment by digorithm** [comment]
Once https://github.com/FuelLabs/fuels-rs/issues/415 is done, the `abigen!` will stop generating structs for `Option`s, avoiding conflicts like this altogether. All references to Sway's `Option<T>` will be actual Rust's `Option<T>` in the generated code. 

The problem you're describing is related to it because the generated `Option`s are conflicting, `Option`s shouldn't be a generated struct, it should be just Rust's `Option<T>`, and that's the intent of https://github.com/FuelLabs/fuels-rs/issues/415.
